### PR TITLE
[xcode11] Remove the WatchKit framework from iOS while keeping API stability. Fixes #6492.

### LINKED ISSUE #6492: [xcode11] The WatchKit framework has been removed from the iOS SDK
The WatchKit framework has been removed from the iOS SDK.

In [Xcode 11 beta 3's release notes][1], Apple explains:

> The WatchKit framework is no longer included in the iOS SDK. If you’re using WatchKit APIs from iOS, you need to remove this use. The WatchKit framework remains available on watchOS. If you’re using WatchKit APIs from iOS to infer availability of features on the paired Apple Watch, include information about your use case when you submit feedback to [Feedback Assistant][2].

Tasks:

* [x] Remove support for linking with the WatchKit framework in mtouch.
* [x] Replace the WatchKit bindings with stubs that throw InvalidPlatformExceptions.

[1]: https://developer.apple.com/documentation/xcode_release_notes/xcode_11_beta_3_release_notes?language=objc
[2]: https://feedbackassistant.apple.com/

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Copy all generated sources and modify them to throw PlatformNotSupported exceptions.
* Adjust some existing source code to also throw PlatformNotSupported exceptions.
* Sprinkle Obsolete attributes generously.
* Stop generating code for the WatchKit framework for iOS.
* Fix mtouch to not link with WatchKit, and show a warning if we detect code
  that want to use WatchKit.
* A few test fixes were also needed.

Fixes https://github.com/xamarin/xamarin-macios/issues/6492.

### CODE REVIEW

File: src/WatchKit/WKDefs.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 namespace WatchKit {
 	[iOS (8,2)]
+	[Unavailable (PlatformName.iOS)]
```
Comment (spouliot) [inline src/WatchKit/WKDefs.cs]: not sure how the IDE will interpret (and show) this mix of `[Introduced]` and `[Available]` ?!?

File: tools/common/Target.cs [hunk 2]
```diff
@@ -205,9 +205,9 @@ public void GatherFrameworks ()
 							}
 							break;
 						case "WatchKit":
-							// Xcode 11 beta 1 doesn't ship WatchKit for iOS
+							// Xcode 11 doesn't ship WatchKit for iOS
 							if (Driver.XcodeVersion.Major == 11 && App.Platform == ApplePlatform.iOS) {
-								ErrorHelper.Warning (99, "Not linking with WatchKit because Xcode 11 beta 1 does not ship with the WatchKit framework.");
+								ErrorHelper.Warning (5219, "Not linking with WatchKit because it has been removed from iOS.");
```
Comment (spouliot) [inline tools/common/Target.cs]: does the condition still makes sense ? as we'll never link with WatchKit anymore (whatever Xcode version is being used)

Comment (rolfbjarne) [reply tools/common/Target.cs]: It shouldn't ever be hit, but _shouldn't_ and _won't_ are quite far apart in the dictionary.

File: docs/website/mtouch-errors.md [hunk 3]
```diff
@@ -2545,6 +2545,14 @@ If this is not the case, please file a [bug report](https://github.com/xamarin/x
 
 The name of an Objective-C protocol can't contain certain characters which means that the `Adopts` attribute on the corresponding managed class can't have the `ProtocolType` parameter containing them. Please refer to the provided error message and fix accordingly.
 
+### MT4178: The class '{class}' will not be registered because the WatchKit framework has been removed from iOS.
+
+Apple removed the WatchKit framework on iOS, so any code that uses it will not work anymore.
```
Comment (spouliot) [inline docs/website/mtouch-errors.md]: That seems inaccurate as introspection did not complain. Only the headers (not the framework) was removed (at least so far)

File: src/WatchKit/WKDefs.cs (hunk 1 above)
Comment (rolfbjarne) [reply src/WatchKit/WKDefs.cs]: The [Unavailable] makes the generator not generate anything (for iOS, still generates for watchOS), so there's no mix in the final source code.

The source code used for iOS is the manual code elsewhere in this PR.

**Comment by rolfbjarne** [comment]
This is how it looks like in the IDE:

Screen Shot 2019-07-15 at 19 37 00

**Comment by rolfbjarne** [comment]
```
08:11:13 sh: /opt/X11/bin/cp: No such file or directory
08:11:13 make[1]: *** [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/builds/install/bcl/monotouch/Facades/System.Reflection.DispatchProxy.dll] Error 127
08:11:13 make[1]: *** Waiting for unfinished jobs....
```

**Comment by rolfbjarne** [comment]
build

**Comment by spouliot** [comment]
https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10714/API_20diff_20_28PR_20only_29/ shows some API are removed (not deprecated)

**Comment by rolfbjarne** [comment]
@spouliot those are all API that were added with Xcode 11, so we don't need them to maintain backwards compatibility.

**Pull request merged**